trace/doc.go

```diff
@@ -0,0 +1,164 @@
+// Package trace provies an API for initiating and reporting SSF
+// traces and attaching spans to them. Veneur's tracing API also
+// provides an experimental opentracing compatibility layer.
+//
+// The Veneur tracing API is completely independent of the opentracing
+// compatibility layer, with the exception of the function
+// StartSpanFromContext.
+//
+// Setup and Structure of the API
+//
+// To use the API, user code must set its service name via the global
+// variable Service, which is used by trace visualization tools to
+// tell apart the spans reported by different services. Code using the
+// trace API must set its Service, and set it only once, in main():
+//
+//    trace.Service = "my-awesome-service"
+//
+// The trace API is built on three abstractions: the
+// Client, the ClientBackend and the Span.
+//
+// Context
+//
+// Since it is very common for networked code to use context.Context,
+// veneur's trace API is designed to be used together with
+// Contexts. The API uses Contexts to store the current Span, and will
+// automatically pick the appropriate parent span from the Context.
+//
+// To effectively use the trace API, user code needs to ensure that
+// all functions that communicate externally (which includes using
+// traces) take Contexts as parameters, and pass them down.
+//
+// See https://blog.golang.org/context for an introduction to using
+// Context.
+//
+// Spans
+//
+// Spans represent the duration of a "unit of work" in a system: They
+// have a start and an end time, a name, and additional data that can
+// help identify what the work was. Example units of work could be the
+// handling of an HTTP request, or a database read. For convenience,
+// it is usually easiest to have a span represent a single function to
+// delineate the start and end, which makes it easy to create the Span
+// and report it via a defer statement.
+//
+// To create a new Span, use StartSpanFromContext. This function will
+// create a trace parented to the traces that came before it in the
```

Comment:
"parented to" is a little clunky. I know what it means, but I'd rather spell it out ("a trace whose parent ID is set to the trace which is in the context passed to StartSpanFromContext").